### PR TITLE
libc/execinfo: add backtrace support based on EABI Unwinder 

### PR DESCRIPTION
**Pull request by anchao**
## Summary

libc/execinfo: add backtrace support based on EABI Unwinder 
libc/execinfo: add dump_stack support

1. Flags < -funwind-tables > is required
2. Keep the section ".exidx" in linker script

## Impact

N/A

## Testing

```
diff --git a/examples/hello/hello_main.c b/examples/hello/hello_main.c
index fcc04112..4b85a41b 100644
--- a/examples/hello/hello_main.c
+++ b/examples/hello/hello_main.c
@@ -24,6 +24,7 @@
 
 #include <nuttx/config.h>
 #include <stdio.h>
+#include <execinfo.h>
 
 /****************************************************************************
  * Public Functions
@@ -36,5 +37,6 @@
 int main(int argc, FAR char *argv[])
 {
   printf("Hello, World!!\n");
+  dump_stack();
   return 0;
 }
```

```
ap> hello
Hello, World!!
[   73.800676] [42] [  INFO] [BackTrace]:   0xc0d9bfe  0xc0f3330  0xc0dba0a   0x210838
```

```
$ arm-none-eabi-addr2line -e nuttx 0xc0d9bfe  0xc0f3330  0xc0dba0a   0x210838
nuttx/libs/libc/debug/lib_dumpstack.c:52
apps/examples/hello/hello_main.c:42
nuttx/libs/libc/sched/task_startup.c:165
nuttx/sched/task/task_start.c:144
```

### CODE REVIEW

File: libs/libc/debug/Kconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# specific implementations.  Not all architectures support implementations
+# for every library function.
+
+menu "Library Debugging"
+
+config EABI_UNWINDER
+	bool "EABI Stack Unwinder"
+	default "n"
+	---help---
+		This option enables stack unwinding support in the nuttx
```
Comment (gustavonihei) [inline libs/libc/debug/Kconfig:16]:
```suggestion
		This option enables stack unwinding support in NuttX
```

File: libs/libc/debug/Kconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		using the information automatically generated by the
+		compiler. The resulting nuttx image is slightly bigger but
```
Comment (gustavonihei) [inline libs/libc/debug/Kconfig:18]:
```suggestion
		compiler. The resulting image is slightly bigger but
```
nit: I think `nuttx` may be omitted here

Comment (acassis) [reply libs/libc/debug/Kconfig:18]: Or at least written correctly: NuttX :-D

Comment (anchao) [reply libs/libc/debug/Kconfig]: Done

File: libs/libc/debug/Kconfig (hunk 1 above)
Comment (anchao) [reply libs/libc/debug/Kconfig]: Done

**Comment by xiaoxiang781216** [comment]
let's ignore:
```
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:38:2: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:53:7: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:54:28: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:64:41: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:74:6: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:74:25: error: Mixed case identifier found
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/libs/libc/debug/lib_backtrace.c:116:2: error: Mixed case identifier found
```
since unwind API is defined by toolchain, we can't change them.

**Comment by davids5** [comment]
@xiaoxiang781216, @anchao - should the allowed words be added to the whitelist as a commit in this PR?

**Comment by xiaoxiang781216** [comment]
@anchao please provide a new patch to update g_white_list in tools/nxstyle.c.

**Comment by anchao** [comment]
Done, @xiaoxiang781216 @davids5  please review again, thanks

**Review by davids5 — APPROVED** [review]
@anchao Thank you!

**Pull request merged**